src/toxcore/tcp/connection_id.rs

```diff
@@ -0,0 +1,92 @@
+//! Connection ID definition.
+
+use toxcore::binary_io::*;
+use toxcore::tcp::links::MAX_LINKS_N;
+
+use nom::be_u8;
+
+/// Connection ID is either a number between [16, 255] or 0. Zero can be
+/// included in a response and means that the previous request was invalid.
+/// If a connection id is a number between [16, 255] it can be uniquely mapped
+/// to a connection index between [0, 239].
+#[derive(Debug, PartialEq, Eq, Clone, Copy)]
+pub struct ConnectionId(u8);
```

Comment:
How about `ConnectionId(Option<u8>)`?

Reply:
It's internal private state so won't change a lot. But it will be less memory efficient because Option<u8> can't be optimized to single u8.

Reply:
We can use Option<NonZeroU8>: https://doc.rust-lang.org/std/num/struct.NonZeroU8.html . But it will change things only inside `ConnectionId` so should we?

Reply:
Oh thanks! `Option<NonZeroU8>` is very idiomatic in our case.

Reply:
Check the last commit. `Option<NonZeroU8>` only complicates the code and adds one more `unwrap`. Do you still think we need it?

Reply:
The API is the same but the inner representation is more idiomatic. I think we should use it.